pdr_backend/lake/duckdb_data_store.py

```diff
@@ -349,3 +352,105 @@ def get_query_move_table_data(self, from_table, to_table):
         return (
             f"INSERT INTO {to_table.table_name} SELECT * FROM {from_table.table_name};"
         )
+
+    @enforce_types
+    def export_tables_to_parquet_files(
+        self,
+        seconds_between_exports: int,
+        number_of_files_after_which_re_export_db: int,
+    ):
+        export_folder_path = get_export_folder_path(self.base_path)
+        tables = self.query_data("SHOW TABLES")["name"]
+
+        # Ensure export folder exists
+        os.makedirs(export_folder_path, exist_ok=True)
+
+        for table in tables:
+            table_folder_path = os.path.join(export_folder_path, table)
+            os.makedirs(table_folder_path, exist_ok=True)
+
+            if self._should_nuke_table_folders_and_re_export_db(
+                table_folder_path, number_of_files_after_which_re_export_db, table
+            ):
+                self._nuke_table_folders_and_re_export_db(table_folder_path)
+
+            if not self._should_export(table_folder_path, seconds_between_exports):
+                continue
+
+            self._export_table_to_parquet(table, table_folder_path)
+
+    def _should_export(
+        self, table_folder_path: str, seconds_between_exports: int
+    ) -> bool:
+        max_timestamp_from_parquet = self._get_max_timestamp_from_parquet_files(
+            table_folder_path
+        )
+        current_timestamp = UnixTimeS(int(time.time())).to_milliseconds()
+
+        return (
+            current_timestamp - max_timestamp_from_parquet
+            >= seconds_between_exports * 1000
+        )
+
+    def _get_max_timestamp_from_parquet_files(self, table_folder_path: str) -> int:
+        try:
+            result = duckdb.execute(
```

Comment:
you can use scalar=True here, it's already implemented

Reply:
It's implemented if we use duckdb data store object to query, if we query files directly using duckdb, it's not implemented